sdk_commons/sdk_commons/config.py

```diff
@@ -0,0 +1,103 @@
+import abc
+from enum import Enum
+
+
+class SDKConfig(abc.ABC):
+    """
+    This is the abstract common interface that every venues
+    are expended to implement. Each method will be invoked by
+    API providing all listed parameters but each venue is free
+    to only include needed parameters in their own concrete
+    implementation. All other parameters will be consumed by
+    *args and **kwargs that are always expected to be included
+    A template venue is also provided as example for a concrete
+    implementation.
+    """
+
+    @property
+    @abc.abstractmethod
+    def authorization_pages(self):
+        """
+        Set this property with a class containing
+        mainnet and testnet properties
+        """
+        pass
+
+    @abc.abstractmethod
+    def venue_chains(self) -> Enum:
+        """
+        Return an Enum listing all supported chains
+        """
+        pass
+
+    @abc.abstractmethod
+    def create_offer(
+        self,
+        address: str,
+        chain_id: int,
+        rpc_uri: str,
+        # TODO: rename into offer_token
+        oToken: str,
+        bidding_token: str,
+        min_price: int,
+        min_bid_size: int,
+        offer_amount: int,
+        public_key: str,
+        private_key: str,
+        *args,
```

Comment:
is *args necessary to add? I think we could have all optionals in the kwargs

Reply:
Good question. Basically I added it to be more flexible and because it costs very few to support both but I don't have a strong opinion on that. Surely kwargs alone already cover near all the requirements. If you prefer a slightly simplified interface we can drop args, for me it is ok

Reply:
I think we could drop that to keep a more concise interface

Reply:
Dropped